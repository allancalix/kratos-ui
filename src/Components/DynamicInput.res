module NonStandardProps = {
  @react.component
  let make = (~props, ~children) => React.cloneElement(children, props)
}

let defaultClasses = "appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm"
let submitClasses = "group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"

module InputField = {
  @react.component
  let make = (~attributes: Kratos.uiNodeInputAttributes, ~placeholder: string="", ~submitButtonLabel=?) => switch attributes.\"type" {
    | "submit" => <button
          className={submitClasses}
          name={attributes.name}
          value={attributes.value -> Belt.Option.getWithDefault("")}
          placeholder={placeholder}
          type_={attributes.\"type"}
          required={attributes.required -> Belt.Option.getWithDefault(false)}>
          {React.string(submitButtonLabel -> Belt.Option.getWithDefault("Submit"))}
          </button>
    | _ => <input
          className={defaultClasses}
          name={attributes.name}
          defaultValue={attributes.value -> Belt.Option.getWithDefault("")}
          placeholder={placeholder}
          type_={attributes.\"type"}
          required={attributes.required -> Belt.Option.getWithDefault(false)}/>
  }
}

@react.component
let make = (~node: Kratos.uiNode) => switch node.meta.label {
  | Some(label) => {
    <>
      <NonStandardProps props={"data-testid": "label"}>
        <label key={Js.Int.toString(label.id)} className="sr-only">
          {React.string(label.text)}
        </label>
      </NonStandardProps>
      <InputField attributes={node.attributes} placeholder={label.text} submitButtonLabel={label.text} />
    </>
  }
  | None => <InputField attributes={node.attributes} />
}
