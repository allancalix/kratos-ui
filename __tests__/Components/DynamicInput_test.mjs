// Generated by ReScript, PLEASE EDIT WITH CARE

import * as Jest from "@glennsl/bs-jest/src/jest.mjs";
import * as Curry from "rescript/lib/es6/curry.js";
import * as React from "react";
import * as JestDom from "bs-jest-dom/src/JestDom.mjs";
import * as DynamicInput from "../../src/Components/DynamicInput.mjs";
import * as ReactTestingLibrary from "@rescriptbr/react-testing-library/src/ReactTestingLibrary.mjs";

Jest.describe("DynamicInputComponent", (function (param) {
        var createComponent = function (param) {
          return React.createElement(DynamicInput.make, {
                      name: "password",
                      type: "password",
                      required: true
                    });
        };
        var createComponentWithLabel = function (param) {
          return React.createElement(DynamicInput.make, {
                      name: "password",
                      type: "password",
                      label: {
                        context: undefined,
                        id: 10,
                        text: "Cool Label",
                        type: "info"
                      },
                      required: true
                    });
        };
        Jest.test("when label is provided then matches expected", (function (param) {
                return Jest.Expect.toMatchSnapshot(Jest.Expect.expect(ReactTestingLibrary.render(undefined, undefined, undefined, undefined, undefined, createComponentWithLabel(undefined)).container));
              }));
        Jest.test("when no label is provided then matches snapshot", (function (param) {
                return Jest.Expect.toMatchSnapshot(Jest.Expect.expect(ReactTestingLibrary.render(undefined, undefined, undefined, undefined, undefined, createComponent(undefined)).container));
              }));
        return Jest.test("when label is provided then has visible label", (function (param) {
                      var partial_arg = {
                        NAME: "Str",
                        VAL: "Cool Label"
                      };
                      var arg = function (param) {
                        return JestDom.toHaveTextContent(partial_arg, param);
                      };
                      var partial_arg$1 = {
                        NAME: "Str",
                        VAL: "label"
                      };
                      var arg$1 = function (param, param$1) {
                        return ReactTestingLibrary.getByTestId(partial_arg$1, param, param$1);
                      };
                      return Curry._2(arg, undefined, expect(Curry._2(arg$1, undefined, ReactTestingLibrary.render(undefined, undefined, undefined, undefined, undefined, createComponentWithLabel(undefined)))));
                    }));
      }));

export {
  
}
/*  Not a pure module */
