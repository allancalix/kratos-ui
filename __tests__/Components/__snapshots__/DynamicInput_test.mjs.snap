// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicInputComponent when label is provided then matches expected 1`] = `
<div>
  <label
    class="sr-only"
    data-testid="label"
  >
    Cool Label
  </label>
  <input
    class="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm"
    data-testid="password"
    name="password"
    placeholder="Cool Label"
    required=""
    type="password"
    value=""
  />
</div>
`;

exports[`DynamicInputComponent when no label is provided then matches snapshot 1`] = `
<div>
  <input
    class="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm"
    data-testid="password"
    name="password"
    placeholder=""
    required=""
    type="password"
    value=""
  />
</div>
`;
